### PR TITLE
[W5][W13-3]Cassey Low

### PR DESCRIPTION
**Pull request by case141**
Added `deletebyname` feature which deletes all people with the names matching the keywords entered.

### CODE REVIEW

**Comment by DoItTomorrow** [comment]
Good feature. Will likely be useful in later versions. Search function for name should be refactored out.

**Comment by suriruhani** [comment]
Great feature to include! Since this action is irreversible, you might want to provide the users an option to review the contacts that are going to be deleted and agree once before the action is carried out.

**Comment by ToonDragon** [comment]
Excellent feature that will definitely see use later on.  A possible improvement could be to allow users to delete all names containing an entered variable all at once.  This might be dangerous but could also be very time saving when removing all entries of a certain type.  

Variables are well named and junctions are properly refactored into their own class.  Very readable and understandable.

**Comment by fterh** [comment]
DeleteByNameCommand.java:
* You should avoid importing `java.util.*`, it pollutes the namespace and may inadvertently cause issues
* Some whitespace formatting issues (e.g. lines 42, 43)

**Comment by case141** [comment]
@fterh Thanks for pointing that out! I probably missed it while I was correcting the errors.

**Comment by case141** [comment]
@DoItTomorrow The search function is needed to retrieve names containing the keywords entered, therefore it is necessary although it can be further simplified later on.

File: docs/UserGuide.adoc [hunk 1]
```diff
@@ -111,6 +111,25 @@ Deletes the 2nd person in the address book.
...
+****
+
+Examples:
+
+* `list` +
+`deletebyname John` +
+Deletes the person(s) containing the name 'John' in the address book.
+
+* `find John Doe` +
+`delete John Doe` +
```
Comment (ewaldhew) [inline docs/UserGuide.adoc]: Mistake in the command here

File: src/seedu/addressbook/commands/DeleteByNameCommand.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        return new HashSet<>(keywords);
+    }
+
+    @Override
+    public CommandResult execute() {
+        final List<ReadOnlyPerson> personsFound = getPersonsWithNameContainingAnyKeyword(keywords);
+        Iterator<ReadOnlyPerson> it = personsFound.iterator();
+        String deletedPersonsList = "";
+        try{
+            for(personsFound.listIterator(); it.hasNext();){
```
Comment (ewaldhew) [inline src/seedu/addressbook/commands/DeleteByNameCommand.java]: No space before the brace :open_mouth:

**Review by ewaldhew — COMMENTED** [review]
Nice enhancement, good job updating the tests and documentation too.

Close PR after reading @case141

**Pull request closed without merging**